src/Subscription.js

```diff
@@ -56,13 +60,26 @@ export default class Subscription extends EventEmitter {
         /** * Message handlers ** */
 
         this.on('unsubscribed', () => {
+            this._clearGaps()
             this.setResending(false)
         })
 
         this.on('disconnected', () => {
             this.setState(Subscription.State.unsubscribed)
+            this._clearGaps()
             this.setResending(false)
         })
+
+        this.on('error', () => {
+            this._clearGaps()
+        })
+    }
+
+    _clearGaps() {
+        Object.keys(this.gaps).forEach((gap) => {
+            clearInterval(gap)
```

Comment:
Should this be `clearInterval(this.gaps[gap])`? (might avoid bugs and be more consistent if you use the variable name `key` instead of `gap`)

Reply:
Plus add test that catches this bug (checks that the resend is not retried if the sub is disconnected or unsubscribed)